drivers/counter/counter_nrfx_timer.c

```diff
@@ -115,11 +125,60 @@ static inline u32_t counter_nrfx_get_cc_value(struct device *dev,
 	return cc_val;
 }
 
+/* Function calculates distance between to values assuming that one first
+ * argument is in front and that values wrap.
+ */
+static u32_t tick_diff(u32_t val, u32_t old, u32_t top)
+{
+	if (top & (top + 1)) {
+		/* if top is not 2^n-1 */
+		u32_t diff;
+
+		diff = (val > old) ? (val - old) : val + top + 1 - old;
+
+		return diff;
+	}
+
+	return (val - old) & top;
+}
+
+static int set_cc(struct device *dev, u8_t chan_id, u32_t val)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_timer_t *tmr = &nrfx_config->timer;
+	NRF_TIMER_Type  *reg = tmr->p_reg;
+	u32_t now;
+	u32_t top = counter_get_top_value(dev);
+	nrf_timer_task_t read_tsk =
+			NRF_TIMER_TASK_CAPTURE0 + sizeof(u32_t)*COUNTER_READ_CC;
+	nrf_timer_event_t evt =
+			NRF_TIMER_EVENT_COMPARE0 + sizeof(u32_t)*chan_id;
+	u32_t int_mask = NRF_TIMER_INT_COMPARE0_MASK << chan_id;
+
+	nrf_timer_cc_write(reg, chan_id, val);
+	nrf_timer_event_clear(reg, evt);
+
+	nrf_timer_task_trigger(reg, read_tsk);
+	now = nrf_timer_cc_read(reg, COUNTER_READ_CC);
+
+	if (tick_diff(val - 1, now, top) > (top - nrfx_config->guard_window)) {
+		WRN("late setting CC%d to %d (now %d)", chan_id, val, now);
+		nrf_timer_event_clear(reg, evt);
+		return -1;
+	} else {
+		DBG("setting CC%d to %d (now %d)", chan_id, val, now);
+	}
+
+	nrf_timer_int_enable(reg, int_mask);
```

Comment:
Where is nrf_timer_int_disable()? If outside of this function I would suggest moving this line there so IRQ status could be easily traced.

Reply:
it's been part of `nrfx_timer_compare_int_disable()`. Note that function for setting CC works directly on hal (nrf_timer) while rest of the driver on the nrfx driver (nrfx_timer). It's been done like that because this late detection is not part of nrfx driver (may be in future extended).